src/main/java/services/gorest/stepdefinition/PostsStepsDefinition.java

```diff
@@ -0,0 +1,94 @@
+package services.gorest.stepdefinition;
+
+
+import cucumber.api.Scenario;
+import cucumber.api.java.After;
+import cucumber.api.java.Before;
+import cucumber.api.java.en.When;
+import io.restassured.response.Response;
+import net.thucydides.core.annotations.Steps;
+import services.gorest.actions.post.CreatePost;
+import services.gorest.actions.post.DeletePost;
+import services.gorest.actions.post.GetPost;
+import services.gorest.actions.post.UpdatePost;
+import services.gorest.actions.user.CreateUser;
+import services.gorest.actions.user.DeleteUser;
+import services.gorest.models.responses.GetPostResponse;
+import services.gorest.models.responses.GetUserResponse;
+import utils.methods.ReusableMethods;
+
+import static utils.variables.SessionVariableManager.getSessionVariable;
+import static utils.variables.SessionVariableManager.setSessionVariable;
+import static utils.variables.SessionVariables.*;
+
+public class PostsStepsDefinition {
+
+    @Steps
+    private CreatePost createPost;
+
+    @Steps
+    private GetPost getPost;
+
+    @Steps
+    private UpdatePost updatePost;
+
+    @Steps
+    private DeletePost deletePost;
+
+    @Steps
+    private ReusableMethods reusableMethods;
+
+    @Steps
+    private CreateUser createUser;
+
+    @Steps
+    private DeleteUser deleteUser;
+
+    @Before("@PostSmoke")
+    public void createPrereq() {
+        Response response = createUser.whenCreateRandomUser();
+        setSessionVariable(VAR_USER_ID, response.as(GetUserResponse.class).getResult().getId());
+    }
+
+    @When("^I create a new post with my user id (.*), I provide the title (.*) and add the following body:$")
+    public void whenCreatePost(String userId, String title, String body) {
+        int uId = Integer.valueOf(reusableMethods.setSessionVariableAsExpected(userId, VAR_USER_ID));
+        Response response = createPost.whenCreateNewPost(uId, title, body);
+        setSessionVariable(VAR_POST_ID, response.as(GetPostResponse.class).getResult().getId());
```

Comment:
In cases like this, just think what should happen in a negative test:
1. try to create a post
2. the post is not created
3. you validate the post isn't created

In #2 this line will return a null pointer. I will recommend to assign the value if not null only